tests/test_rows.py

```diff
@@ -102,6 +102,12 @@ def test_kwargs_row(conn):
     assert p.age == 42
 
 
+def test_scalar_row(conn):
+    cur = conn.cursor(row_factory=rows.scalar_row)
+    cur.execute("select 1")
+    assert cur.fetchone() == 1
```

Comment:
The test should include a case to show that the error is raised (with a query returning 0 columns, and with a query returning >1 column if we think that's the way to go).